pkgs/servers/sql/postgresql/ext/vectorchord/package.nix

```
@@ -0,0 +1,83 @@
+{
+  buildPgrxExtension,
+  cargo-pgrx_0_12_9,
+  clang_18,
+  fetchFromGitHub,
+  lib,
+  nix-update-script,
+  nixosTests,
+  postgresql,
+  replaceVars,
+  rustPlatform,
+}:
+let
+  # Upstream recommends clang 18
+  clang = clang_18;
+  rustPlatform' = rustPlatform // {
+    bindgenHook = rustPlatform.bindgenHook.override { inherit clang; };
+  };
+in
+(buildPgrxExtension.override {
+  # Upstream only works with a fixed version of cargo-pgrx for each release,
+  # so we're pinning it here to avoid future incompatibility.
+  cargo-pgrx = cargo-pgrx_0_12_9;
+  rustPlatform = rustPlatform';
+})
+  rec {
+    inherit postgresql;
+
+    pname = "vectorchord";
+    version = "0.2.2";
+
+    patches = [
+      # Tell the `simd` crate to use the flags from the rust bindgen hook
+      (replaceVars ./0001-read-clang-flags-from-environment.diff {
+        clang = lib.getExe clang;
+      })
+      # Add feature flags needed for features not yet stabilised in rustc stable
+      ./0002-add-feature-flags.diff
+    ];
+
+    src = fetchFromGitHub {
+      owner = "tensorchord";
+      repo = "vectorchord";
+      tag = version;
+      hash = "sha256-hZnB/c38PbJBcrd3N6N5IKkmcgG8NE0Xvxb0G5kbn1E=";
+    };
+
+    useFetchCargoVendor = true;
+    cargoHash = "sha256-lob66NsflCAE14KBDp56PugkoisDSS4NWMRPBbBAmkA=";
+
+    # Set appropriate version on vchord.control, otherwise it won't show up on PostgreSQL
+    postPatch = ''
+      substituteInPlace ./vchord.control --subst-var-by CARGO_VERSION ${version}
+    '';
+
+    # Include upgrade scripts in the final package
+    # https://github.com/tensorchord/VectorChord/blob/0.2.2/.github/workflows/release.yml#L85
+    postInstall = ''
+      cp sql/upgrade/* $out/share/postgresql/extension/
+    '';
+
+    env = {
+      # Bypass rust nightly features not being available on rust stable
+      RUSTC_BOOTSTRAP = 1;
+    };
+
+    # This crate does not have the "pg_test" feature
+    usePgTestCheckFeature = false;
+
+    passthru = {
+      updateScript = nix-update-script { };
+      tests = nixosTests.postgresql.vectorchord.passthru.override postgresql;
+    };
+
+    meta = {
+      description = "Scalable, fast, and disk-friendly vector search in Postgres, the successor of pgvecto.rs. ";
```

```suggestion
      description = "Scalable, fast, and disk-friendly vector search in Postgres, the successor of pgvecto.rs";
```
See contributing guidelines.